File: examples/canonical-azure-functions/blob-trigger-receipt-ocr/python/tests/integration/lane-a-miniblue/test_handler.py
import pathlib
import sys

root = pathlib.Path(__file__).resolve().parents[6]
sys.path.insert(0, str(root))

from examples.canonical_azure_functions_import import load_python_handler

handler = load_python_handler(
    "examples/canonical-azure-functions/blob-trigger-receipt-ocr/python/src/handler.py"
)
result = handler.process_blob({"name": "lane-a/receipt.pdf", "version": "azurite-1", "total": 12.5})
assert result["cosmos_document"]["id"] == result["idempotency_key"]
print("lane-a-miniblue blob receipt passed")

File: examples/canonical-azure-functions/blob-trigger-receipt-ocr/typescript/tests/integration/lane-a-miniblue/test_handler.js
const assert = require("node:assert/strict");
const { processBlob } = require("../../../src/handler.js");

const result = processBlob({ name: "lane-a/receipt.pdf", version: "azurite-etag", total: 12.5 });
assert.equal(result.cosmos_document.id, result.idempotency_key);
console.log("lane-a-miniblue blob receipt ocr typescript passed");
